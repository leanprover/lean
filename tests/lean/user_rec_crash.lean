example (a b : Prop) (h : a ∧ b) : a :=
begin
  induction h using and.rec_on with h₁ h₂,
end
