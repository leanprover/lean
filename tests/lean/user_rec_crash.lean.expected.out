user_rec_crash.lean:3:2: error:invalid 'induction' tactic, invalid user defined recursor, 'and.rec_on' does not support dependent elimination, and position of the major premise was not specified (solution: set attribute '[recursor <pos>]', where <pos> is the position of the major premise)
proof state:
a b : Prop,
h : a ∧ b
⊢ a
user_rec_crash.lean:4:0: error: don't know how to synthesize placeholder
a b : Prop,
h : a ∧ b
⊢ a
user_rec_crash.lean:4:0: error: failed to add declaration '14.0' to environment, value has metavariables
remark: set 'formatter.hide_full_terms' to false to see the complete term
  λ (a b : Prop) (h : …),
    ?M_1
